src/services/dexes/sovryn/SovrynMultiProcessor.ts

```diff
@@ -3,14 +3,39 @@ import {inject, injectable} from 'inversify';
 import {SovrynPriceFetcher, PairRequest} from './SovrynPriceFetcher.js';
 import {FeedFetcherInterface, FetcherName} from '../../../types/fetchers.js';
 import {FeedFetcher} from '../../../types/Feed.js';
+import {PriceDataRepository} from '../../../repositories/PriceDataRepository.js';
 
 @injectable()
 export default class SovrynMultiProcessor implements FeedFetcherInterface {
   @inject(SovrynPriceFetcher) sovrynFetcher!: SovrynPriceFetcher;
+  @inject(PriceDataRepository) private priceDataRepository!: PriceDataRepository;
 
   async apply(feedFetchers: FeedFetcher[]): Promise<(number | undefined)[]> {
     const request = this.createRequest(feedFetchers);
-    return await this.sovrynFetcher.apply(request);
+    const prices = await this.sovrynFetcher.apply(request);
+
+    const symbolToBaseAndQuote = (symbol: string) => {
+      const symbols = symbol.split('-');
+      return symbols.length != 2 ? ['not-found', 'not-found'] : symbols;
+    };
+
+    for (const [ix, price] of prices.entries()) {
+      const [feedBase, feedQuote] = symbolToBaseAndQuote(feedFetchers[ix].symbol || '-');
+
+      if (price) {
+        await this.priceDataRepository.savePrice({
+          fetcher: FetcherName.SOVRYN_PRICE,
+          value: price.toString(),
+          valueType: 'string',
+          timestamp: Date.now(), // prices coming from SovrynFetcher don't contain any timestamp
+          feedBase,
+          feedQuote,
+          fetcherSource: 'Sovryn Protocol',
```

Comment:
it should be static property of the class

Reply:
done